### PR TITLE
Fix adaptive_max_pool2d fallback

### LINKED ISSUE #136332: LoweringException: NameError: name 'max_pool2d_with_indices' is not defined
### 🐛 Describe the bug

Regression compared to PyTorch 2.3.1, trying to compile a network using ``F.adaptive_max_pool2d`` results in ``LoweringException: NameError: name 'max_pool2d_with_indices' is not defined``. 

Note that ``F.adaptive_avg_pool2d`` works correctly.

```
  File "/home/morbo/git/sjeng/train/venv241/lib/python3.10/site-packages/torch/_inductor/graph.py", line 979, in call_
function
    out = lowerings[target](*args, **kwargs)
  File "/home/morbo/git/sjeng/train/venv241/lib/python3.10/site-packages/torch/_inductor/lowering.py", line 304, in wr
apped
    out = decomp_fn(*args, **kwargs)
  File "/home/morbo/git/sjeng/train/venv241/lib/python3.10/site-packages/torch/_inductor/lowering.py", line 4231, in a
daptive_max_pool2d
    return max_pool2d_with_indices(x, kernel_size)  # type: ignore[name-defined]   # noqa: F821
torch._dynamo.exc.BackendCompilerFailed: backend='compile_fn' raised:
LoweringException: NameError: name 'max_pool2d_with_indices' is not defined
  target: aten.adaptive_max_pool2d.default
  args[0]: TensorBox(StorageBox(
    ComputedBuffer(name='buf98', layout=FixedLayout('cuda', torch.float16, size=[512, 512, 8, 8], stride=[32768, 1, 40
96, 512]), data=Pointwise(
      'cuda',
      torch.float16,
      def inner_fn(index):
          i0, i1, i2, i3 = index
          tmp0 = ops.load(buf90, i1 + 512 * i3 + 4096 * i2 + 32768 * i0)
          tmp1 = ops.to_dtype(tmp0, torch.float32, src_dtype=torch.float16)
          tmp2 = ops.load(buf94, i1)
          tmp3 = tmp1 - tmp2
          tmp4 = ops.load(buf95, i1)
          tmp5 = ops.constant(0, torch.float32)
          tmp6 = ops.constant(32768, torch.float32)
          tmp7 = ops.constant(0, torch.float32)
          tmp8 = tmp6 - tmp5
          tmp9 = ops.maximum(tmp7, tmp8)
          tmp10 = tmp4 / tmp9
          tmp11 = ops.constant(1e-05, torch.float32)
          tmp12 = tmp10 + tmp11
          tmp13 = ops.rsqrt(tmp12)
          tmp14 = tmp3 * tmp13
          tmp15 = ops.load(primals_23, i1)
          tmp16 = tmp14 * tmp15
          tmp17 = ops.load(primals_24, i1)
          tmp18 = tmp16 + tmp17
          tmp19 = ops.to_dtype(tmp18, torch.float16, src_dtype=torch.float32)
          return tmp19
      ,
      ranges=[512, 512, 8, 8],
      origin_node=convert_element_type_23,
      origins={convert_element_type_23, add_40, mul_55, convert_ele...
    ))
  ))
  args[1]: [1, 1]

While executing %submod_1 : [num_users=2] = call_module[target=submod_1](args = (%getitem, %getitem_1, %getitem_2), kw
args = {})
Original traceback:
None
```

Example code:

```
class GlobalPoolingLayer(nn.Module):
    def __init__(self, channels):
        super(GlobalPoolingLayer, self).__init__()
        self.CHANNELS = channels

    def forward(self, x):
        # Mean of each channel
        mean_pool = F.adaptive_avg_pool2d(x, 1)  # b x c
        mean_pool = mean_pool.view(-1, self.CHANNELS)

        # Maximum of each channel
        max_pool = F.adaptive_max_pool2d(x, 1)  # b x c
        max_pool = max_pool.view(-1, self.CHANNELS)

        # Concatenate the results
        return torch.cat((mean_pool, max_pool), dim=1)  # b x (2c)
```

### Versions

Collecting environment information...                                                                        [48/1887]
PyTorch version: 2.4.1+cu124
Is debug build: False
CUDA used to build PyTorch: 12.4
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.5 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: 14.0.0-1ubuntu1.1
CMake version: version 3.22.1
Libc version: glibc-2.35

Python version: 3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-6.5.0-27-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration:
GPU 0: NVIDIA GeForce RTX 4070 SUPER
GPU 1: NVIDIA GeForce RTX 3080 Ti

Nvidia driver version: 560.35.03
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.9.7
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.9.7
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
CPU:                                                                                                         [14/1887]
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Address sizes:                      43 bits physical, 48 bits virtual
Byte Order:                         Little Endian
CPU(s):                             24
On-line CPU(s) list:                0-23
Vendor ID:                          AuthenticAMD
Model name:                         AMD Ryzen 9 3900X 12-Core Processor
CPU family:                         23
Model:                              113
Thread(s) per core:                 2
Core(s) per socket:                 12
Socket(s):                          1
Stepping:                           0
Frequency boost:                    enabled
CPU max MHz:                        4672.0698
CPU min MHz:                        2200.0000
BogoMIPS:                           7585.97
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mm
x fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apic
id aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand la
hf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_co
re perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba ibpb stibp vmmcall fsgsbase bmi1 avx2 smep
 bmi2 cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 cqm_llc cqm_occup_llc cqm_mbm_total cqm
_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid d
ecodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif v_spec_ctrl umip rdpid overflow_recov succor smca sev s
ev_es
Virtualization:                     AMD-V
L1d cache:                          384 KiB (12 instances)
L1i cache:                          384 KiB (12 instances)
L2 cache:                           6 MiB (12 instances)
L3 cache:                           64 MiB (4 instances)
NUMA node(s):                       1
NUMA node0 CPU(s):                  0-23
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Mitigation; untrained return thunk; SMT enabled with STIBP protection
Vulnerability Spec rstack overflow: Mitigation; Safe RET
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Retpolines, IBPB conditional, STIBP always-on, RSB filling, PBRSB-eIBR
S Not affected
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected

Versions of relevant libraries:
[pip3] numpy==1.26.3
[pip3] pytorch-ranger==0.1.1
[pip3] torch==2.4.1+cu124
[pip3] torch-optimizer==0.3.0
[pip3] torchaudio==2.4.1+cu124
[pip3] torchvision==0.19.1+cu124
[pip3] triton==3.0.0
[conda] Could not collect

cc @ezyang @chauhang @penguinwu

### PR DESCRIPTION
**Pull request by isuruf**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #136367

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

Fixes #136332

### CODE REVIEW

File: torch/_inductor/lowering.py [hunk 1]
```diff
@@ -4276,10 +4276,11 @@ def adaptive_max_pool2d(x, output_size):
         return empty(o_size, dtype=x.get_dtype(), device=x.get_device()), empty(
             o_size, dtype=torch.int64, device=x.get_device()
         )
+
     if h_in % h_out == 0 and w_in % w_out == 0:
         kernel_size = [h_in // h_out, w_in // w_out]
         if should_fallback_max_pool2d_with_indices(kernel_size, dilation=[1, 1]):
-            return max_pool2d_with_indices(x, kernel_size)  # type: ignore[name-defined]   # noqa: F821
+            return aten.max_pool2d_with_indices(x, kernel_size)  # type: ignore[name-defined]   # noqa: F821
```
Comment (isuruf) [inline torch/_inductor/lowering.py:4283]: @amjames, should we just call this unconditionally?

**Review by amjames — APPROVED** [review]
This one was my fault. Thanks for fixing it!

File: torch/_inductor/lowering.py (hunk 1 above)
Comment (amjames) [reply torch/_inductor/lowering.py]: FWIW I think using a properly defined fallback handler would work here too

**Review by eellison — APPROVED** [review]

**Pull request closed without merging**